message.d.ts

```diff
@@ -0,0 +1,588 @@
+     * @param messageId For response messages, the message id or '' if no message id is specified. For non-response messages, 0.
+     * @param suggestedPivot The field number at which to start putting fields into the extension object. This is only used if data does not contain an extension object already. -1 if no extension object is required for this message type.
+     * @param repeatedFields The message's repeated fields.
+     * @param opt_oneofFields The fields belonging to each of the message's oneof unions.
+     */
+    static initialize(msg: Message, data: any[] | undefined | null, messageId: string | number, suggestedPivot: number, repeatedFields: number[] | null, opt_oneofFields?: number[][] | null): void;
+    /**
+     * Converts a JsPb repeated message field into an object list.
+     * @param field The repeated message field to be converted.
+     * @param toObjectFn The toObject function for this field.  We need to pass this for effective dead code removal.
+     * @param opt_includeInstance Whether to include the JSPB instance for transitional soy proto support: http://goto/soy-param-migration
+     * @return An array of converted message objects.
+     */
+    static toObjectList<T = any>(field: (T | null)[], toObjectFn: ((() => (object | null)) | null) | (() => (object | null)) | null, opt_includeInstance?: boolean): (object | null)[];
+    /**
+     * Adds a proto's extension data to a Soy rendering object.
+     * @param proto The proto whose extensions to convert.
+     * @param obj The Soy object to add converted extension data to.
+     * @param extensions The proto class' registered extensions.
+     * @param getExtensionFn The proto class' getExtension function. Passed for effective dead code removal.
+     * @param opt_includeInstance Whether to include the JSPB instance for transitional soy proto support: http://goto/soy-param-migration
+     */
+    static toObjectExtension(proto: Message, obj: object, extensions: object, getExtensionFn: (this: any) => any, opt_includeInstance?: boolean): void;
+    /**
+     * Writes a proto's extension data to a binary-format output stream.
+     * @param proto The proto whose extensions to convert.
+     * @param writer The binary-format writer to write to.
+     * @param extensions The proto class' registered extensions.
+     * @param getExtensionFn The proto class' getExtension function. Passed for effective dead code removal.
+     */
+    static serializeBinaryExtensions(proto: Message, writer: any, extensions: object, getExtensionFn: (this: Message | null) => any): void;
+    /**
+     * Reads an extension field from the given reader and, if a valid extension,
+     * sets the extension value.
+     * @param msg A jspb proto.
+     * @param extensions The extensions object.
+     */
+    static readBinaryExtension(msg: Message, reader: BinaryReader, extensions: object, getExtensionFn: (this: Message | null) => any, setExtensionFn: (this: Message | null) => any): void;
```

Comment:
`getExtensionInfo: (this: Message, fieldInfo: any) => any`
`setExtensionInfo: (this: Message, fieldInfo: any, value: any) => Message`